curvefs/src/tools/curvefs_tool.h

```diff
@@ -92,8 +107,31 @@ class CurvefsToolRpc : public CurvefsTool {
     }
 
  protected:
-    virtual void SendRequestToService() {
-        service_stub_func_(controller_.get(), request_.get(), response_.get());
+    /**
+     * @brief send request to host in hostsAddressStr_
+     *
+     * @return true
+     * @return false
+     * @details
+     * as long as one succeeds, it returns true and ends sending
+     */
+    virtual bool SendRequestToServices() {
+        for (const std::string& host : hostsAddressStr_) {
+            if (channel_->Init(host.c_str(), nullptr) != 0) {
+                std::cerr << "Fail init channel to host: " << host << std::endl;
+                continue;
+            }
+            // if service_stub_func_ does not assign a value
+            // it will crash in there
+            service_stub_func_(controller_.get(), request_.get(),
+                               response_.get());
+            if (AfterSendRequestToService(host) == true) {
+                return true;
```

Comment:
if RPC succeed, it's better to save current working host.
for example, another tool needs to communicate with mds in different steps, the first time you should try all MDS until success, but next time, you can directly send RPC to current working one.

Reply:
> if RPC succeed, it's better to save current working host. for example, another tool needs to communicate with mds in different steps, the first time you should try all MDS until success, but next time, you can directly send RPC to current working one.

It's really a good idea, but the command will over after running, and loop input is not accepted.